media/open_zim.ksy

```diff
@@ -0,0 +1,356 @@
+    type: u8
+    doc: position of the directory pointerlist ordered by URL
+  - id: title_pointer_list_ptr
+    -orig-id: titlePtrPos
+    type: u8
+    doc: position of the directory pointerlist ordered by Title
+  - id: cluster_pointer_list_ptr
+    -orig-id: clusterPtrPos
+    type: u8
+    doc: position of the cluster pointer list
+  - id: mime_list_ptr
+    -orig-id: mimeListPos
+    type: u8
+    doc: |
+      position of the MIME type list (also header size)
+      The MIME type list always follows directly after the header, so the mimeListPos also defines the end and size of the ZIM file header.
+      The MIME types in this list are zero terminated strings.
+  - id: main_page_idx
+    -orig-id: mainPage
+    type: u4
+    doc: main page or 0xffffffff if no main page
+  - id: layout_page_idx
+    -orig-id: layoutPage
+    type: u4
+    doc: layout page or 0xffffffffff if no layout page
+  - id: md5_ptr
+    -orig-id: checksumPos
+    type: u8
+    doc: pointer to the md5 checksum of this file without the checksum itself. This points always 16 bytes before the end of the file.
+instances:
+  mime_list:
+    pos: mime_list_ptr
+    type: str_list
+  url_pointer_list:
+    doc: |
+      The URL pointer list is a list of 8 byte offsets to the directory entries.
+      The directory entries are always ordered by URL. Ordering is simply done by comparing the URL strings.
+      Since directory entries have variable sizes this is needed for random access.
+      Zimlib caches directory entries and references the cached entries via the URL pointers.
+    pos: url_pointer_list_ptr
+    type: directory_entry_ptr
+    repeat: expr
+    repeat-expr: article_count
+  title_pointer_list:
+    doc: |
+      The title pointer list is a list of article indices ordered by title. The title pointer list actually points to entries in the URL pointer list. Note that the title pointers are only 4 bytes. They are not offsets in the file but article numbers. To get the offset of an article from the title pointer list, you have to look it up in the URL pointer list.
+      The indirection from titles via URLs to directory entries has two reasons: the pointer list is only half in size as 4 bytes are enough for each entry accessing directory entries by title also makes use of cached directory entries which are referenced by the URL pointers, as implemented in zimlib.
+    pos: title_pointer_list_ptr
+    type: title_index
+    repeat: expr
+    repeat-expr: article_count
+  cluster_pointer_list:
+    pos: cluster_pointer_list_ptr
+    type: cluster_ptr
+    repeat: expr
+    repeat-expr: cluster_count
+  main_page:
+    value: url_pointer_list[main_page_idx]
+    if: main_page_idx != 0xffffffff
+  layout_page:
+    value: url_pointer_list[layout_page_idx]
+    if: layout_page_idx != 0xffffffff
+  md5:
+    pos: md5_ptr
+    type: md5
+  minimal_xz_lzma_size:
+    value: 32
+types:
+  md5:
+    seq:
+      - id: data
+        size: 16
+  uuid:
```

Comment:
Note that the [devel Web IDE](https://ide.kaitai.io/devel/) supports UUID formatting using the `-webide-representation` key, see https://github.com/kaitai-io/kaitai_struct_webide/commit/7f8e6a91bcfa9af3cb84ba2d9f5b452b1f321e1a:

> **Closes kaitai-io/kaitai_struct_webide#55: implement "uuid" formatter for webide-representation**
> 
> Use `-webide-representation: "{field_name:uuid}"` for the standard big-endian format or `{field_name:uuid=ms}` for Microsoft's implementation

I have no idea which UUID format (big-endian or the Microsoft one) the OpenZIM uses, though. The byte order of both formats can be seen on line [`src/utils.ts:84`](https://github.com/kaitai-io/kaitai_struct_webide/blob/7f8e6a91bcfa9af3cb84ba2d9f5b452b1f321e1a/src/utils.ts#L84).